### PR TITLE
Project Mongo Api

### PR DESCRIPTION
**Pull request by AnnikaSonnek**

### CODE REVIEW

**Review by PerErikAndreas — COMMENTED** [review]
Hello Annika!

The task this week was to:
✓ A minimum of one endpoint to return a **collection** of results (array of elements).
✓ A minimum of one endpoint to return a **single** result (single element).
✓ Your API should make use of Mongoose models to model your data and use these models to fetch data from the database.
✓ Your API should be RESTful.

You manage to do all of these plus some extra endpoints. Great work as always! I will leave some comments in the code! 

(I know that the deployed API was up and running last week so I know that it worked but for some reason its down at the moment. )

**Review by PerErikAndreas — COMMENTED** [review]
Hello Annika!

The task this week was to:
✓ A minimum of one endpoint to return a **collection** of results (array of elements).
✓ A minimum of one endpoint to return a **single** result (single element).
✓ Your API should make use of Mongoose models to model your data and use these models to fetch data from the database.
✓ Your API should be RESTful.

You manage to do all of these plus some extra endpoints. Great work as always! I will leave some comments in the code! 

(I know that the deployed API was up and running last week so I know that it worked but for some reason its down at the moment. )

File: server.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+const { Schema } = mongoose;
+dotenv.config()
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+const mongoUrl = process.env.MONGO_URL || "mongodb://127.0.0.1:27017/project-mongo"
```
Comment (PerErikAndreas) [inline server.js:9]: A followed your tread on slack and great that you got it to work. As im on a mac this was the only thing that I did not need to change. "mongodb://127.0.0.1:27017/project-mongo". So these numbers make it possible to see your work locally right?

File: server.js [hunk 2]
```diff
@@ -24,11 +21,162 @@ const app = express();
...
+  seedDatabase()
+  //The function is invoked.
+}
+
+
+//////////////ROUTES//////////////////////
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const bookAPIGuide = {
```
Comment (PerErikAndreas) [inline server.js:61]: Nice and easy to see what endpoints you can use!

File: server.js [hunk 3]
```diff
@@ -24,11 +21,162 @@ const app = express();
...
+      success: false,
+      body: {
+        message: e
+      }
+    })
+  }
+  
+})
+
+app.get("/books/id/:id", async (req, res) => {
```
Comment (PerErikAndreas) [inline server.js:117]: You have handled all the eventual errors from what I can see! Nice!

File: server.js [hunk 4]
```diff
@@ -24,11 +21,162 @@ const app = express();
...
+app.get("/books", async (req, res) => {
+  const {title, authors} = req.query;
+  // we define that we can search for title and authors, seperately or combined. 
+  const response = {
+    success: true, 
+    body: {} // body is an empty object that will hold the response data.
+  }
+  // response is an object that we use to hold the response data that we want to send back to the client in the endpoint /books. 
+
+  const authorsRegex = new RegExp(authors, "i")
```
Comment (OscarSindemark) [inline server.js:86]: I did not follow along with the coding session when they did this, and will be following up in my own code. Seems to make it easier for finding and replacing, as well as splicing!

File: server.js (hunk 2 above)
Comment (OscarSindemark) [inline server.js:61]: I totally ripped this off from you on your expressAPI project! I think it makes it look really nice

**Review by OscarSindemark — APPROVED** [review]
As Andreas has said, you got all the endpoints done and your code is easy to follow!

**Pull request closed without merging**